scene/resources/shape_3d.cpp

```diff
@@ -66,45 +66,105 @@ void Shape3D::set_margin(real_t p_margin) {
 	PhysicsServer3D::get_singleton()->shape_set_margin(shape, margin);
 }
 
-Ref<ArrayMesh> Shape3D::get_debug_mesh() {
-	if (debug_mesh_cache.is_valid()) {
-		return debug_mesh_cache;
+Ref<ArrayMesh> Shape3D::get_debug_mesh(const Color& p_debug_color, const bool p_with_shape_faces) {
```

Comment:
```suggestion
Ref<ArrayMesh> Shape3D::get_debug_mesh(const Color& p_debug_color, bool p_with_shape_faces) {
```
No `const` for simple types, and match header